app/MMB/src/scenes/tripServices/insurance/insuranceSelectionScene/variantButtons/Price.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import * as React from 'react';
+import {
+  Price as OriginalPrice,
+  type StylePropType,
+  Text,
+} from '@kiwicom/mobile-shared';
+import { Translation } from '@kiwicom/mobile-localization';
+
+export type PriceType = {|
+  +amount: ?number,
+  +currency: ?string,
+|};
+
+type Props = {|
+  +price: ?PriceType,
+  +isSelected: boolean,
+  +textDisabled: StylePropType,
+|};
+
+const Price = (props: Props) => {
+  const { price, isSelected, textDisabled } = props;
+  const style = isSelected ? undefined : textDisabled;
+  if (price === null) {
+    return (
+      <Text style={style}>
+        <Translation id="mmb.trip_services.insurance.selection.free" />
+      </Text>
+    );
+  } else if (price && price.amount != null && price.currency != null) {
+    return (
+      <OriginalPrice
+        amount={price.amount}
+        currency={price.currency}
+        style={style}
+      />
+    );
+  }
+  return (
+    <Text style={style}>
+      <Translation passThrough="N/A" />
+    </Text>
```

Comment:
Well, I don't know. It's just that I return `null` in the GQL server when the price is _Free_ for no insurance. Maybe I should return a non-null price that has amount 0. In which case, it makes sense to display free. 
I'm just trying to code defensively in case the GQL was changed for any reason (in the generated types, we get 
``` 
price: ?{ ... },
```
so actually I should have done this better in the GQL server. Thoughts?